src/Extensions/Diagnostics.HealthChecks/ServiceCollectionExtensions.cs

```diff
@@ -30,20 +31,47 @@ public static IHealthChecksBuilder AddServiceFabricHealthChecks(this IServiceCol
 				});
 
 			serviceCollection.TryAddSingleton<ObjectPoolProvider, DefaultObjectPoolProvider>();
+		}
 
-			return serviceCollection
-				.AddHealthCheckPublisher<ServiceFabricHealthCheckPublisher>()
-				.AddHealthChecks();
+		/// <summary>
+		/// Register publisher for processing health check results directly to replicas
+		/// </summary>
+		public static IHealthChecksBuilder AddServiceFabricHealthChecks(this IServiceCollection serviceCollection)
+		{
+			return serviceCollection.AddServiceFabricHealthChecks<ServiceFabricHealthCheckPublisher>();
+		}
+
+		/// <summary>
+		/// Register publisher for processing health check results directly to nodes using REST api
+		/// </summary>
+		public static IHealthChecksBuilder AddRestHealthChecks(this IServiceCollection serviceCollection)
+		{
+			serviceCollection.TryAddSingleton(
+				provider =>
+				{
+					return provider.GetRequiredService<IOptions<RestHealthCheckPublisherOptions>>().Value;
+				});
+			return serviceCollection.AddServiceFabricHealthChecks<RestHealthCheckPublisher>();
 		}
 
 		/// <summary>
 		/// Register publisher for processing health check results
 		/// </summary>
+		public static IHealthChecksBuilder AddServiceFabricHealthChecks<TPublisher>(this IServiceCollection serviceCollection)
+				where TPublisher : class, IHealthCheckPublisher
+		{
+			serviceCollection.ConfigureService();
+			return serviceCollection.AddHealthCheckPublisher<TPublisher>().AddHealthChecks();
+		}
+
+		/// <summary>
+		/// Register publisher using Dependency Injection
+		/// </summary>
 		public static IServiceCollection AddHealthCheckPublisher<TPublisher>(this IServiceCollection serviceCollection)
 			where TPublisher : class, IHealthCheckPublisher
 		{
 			serviceCollection.TryAddEnumerable(ServiceDescriptor.Singleton<IHealthCheckPublisher, TPublisher>());
 			return serviceCollection;
-		}
+		}	
```

Comment:
nit: Trailing whitespace
```suggestion
		}
```